qualcommax: ipq807x: use ascii-env driver for Linksys MX devices

testuser7 opened the pull request:
ascii-env driver allows reading mac addresses directly from devinfo partition from dts level.

Additionally label mac address have been set.

testuser7 committed:
qualcommax: ipq807x: use ascii-env driver for Linksys MX devices

ascii-env driver allows reading mac addresses directly from devinfo partition from dts level.

Additionally label mac address have been set.

Signed-off-by: Paweł Owoc <frut3k7@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/18759
Signed-off-by: Robert Marko <robimarko@gmail.com>

georgemoussalem commented:
> @georgemoussalem I checked and it seems to me that we do not need to set additionally:
> 
> ```
> 					#address-cells = <1>;
> 					#size-cells = <1>;
> ```
> 
> for compatible `ascii-eq-delim-env`.

great, thanks for checking. No build warnings either I assume (#nvmem-cells takes care of the cell count in the phandles to the mac-base node)?
Will remove them as part of a future PR

Ansuel commented:
thanks for the new conversion... May I ask why label-mac-device is needed?

testuser7 commented:
> thanks for the new conversion... May I ask why label-mac-device is needed?

Can be used to set MAC addresses for wireless interfaces for MX4200v2 and MX4300.
For the rest it is not needed. Should I remove it?

There is also a solution proposed by @georgemoussalem but I don't know if it is ready to set multiple mac addresses for one radio definition: https://github.com/openwrt/openwrt/pull/18297

testuser7 committed:
qualcommax: ipq807x: use label MAC to set addresses for wireless interfaces

Use label MAC to set addresses for wireless interfaces
for Linksys MX4200v2 and MX4300.

Signed-off-by: Paweł Owoc <frut3k7@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/18759
Signed-off-by: Robert Marko <robimarko@gmail.com>

testuser7 commented:
@Ansuel Corrected.

georgemoussalem commented:
> > thanks for the new conversion... May I ask why label-mac-device is needed?
> 
> Can be used to set MAC addresses for wireless interfaces for MX4200v2 and MX4300. For the rest it is not needed. Should I remove it?
> 
> There is also a solution proposed by @georgemoussalem but I don't know if it is ready to set multiple mac addresses for one radio definition: #18297

I would not know how to set multiple mac addresses on multi-phy radios, I'm afraid. But definitely worth looking into when I get my hands on an ipq807x or similar device.

testuser7 commented:
Does anything else need to be done to merge these changes?

testuser7 commented:
@Ansuel, @robimarko can we move forward?

robimarko commented:
Thanks! Rebased on top of main and merged!
